modules/data.atmosphere/R/check_met_input.R

```diff
@@ -0,0 +1,97 @@
+#' Check a meteorology data file for compliance with the PEcAn standard
+#'
+#' @param metfile Path of met file to check, as a scalar character.
+#' @param variable_table `data.frame` linking standard names to their
+#'   units. Defaults to [pecan_standard_met_table].
+#' @return `TRUE` (invisibly) if successful, otherwise throw an informative error.
+#' @author Alexey Shiklomanov
+#' @export
+check_met_input_file <- function(metfile, variable_table = pecan_standard_met_table) {
+
+  metfile <- normalizePath(metfile)
+
+  PEcAn.logger::severeifnot(
+    file.exists(metfile),
+    msg = glue::glue("File '{metfile}' does not exist.")
+  )
+
+  nc <- ncdf4::nc_open(metfile)
+
+  test_dims <- tryCatch(
+    testthat::test_that(
```

Comment:
Good catch. I assumed it was already being imported, but it was in-fact in Suggests. I changed this accordingly.

Reply:
Thanks for taking this on Alexy ... But, would it be more appropriate to use assertions rather than tests here? Eg https://cran.r-project.org/web/packages/assertthat/README.html

Reply:
I was considering using `assertthat` (, but I was concerned that if I introduced another dependency, @serbinsh would actually murder me.

More seriously, besides introducing another dependency, one advantage of `testthat` over `assertthat` is that `testthat` expectations, when run inside of a `test_that` block, proceed with evaluation, whereas `assertthat::assert_that` is like `stopifnot` in that it fails on the first failing assertion. This is good here because it allows you to diagnose multiple problems at the same time -- e.g. if both `latitude` and `longitude` have the wrong unit, this code should spit out both errors, whereas using `assert_that`, you would discover one error, then fix the file, then return only to discover the second error, so you'd have to go back _again_ and fix that too (etc., etc.).

Reply:
Case in point:

```r
metfile <- "~/Downloads/nobackup/CRUNCEP.2004.nc"
check_met_input_file(metfile)
# Test failed: 'Dimensions are correct.'
# * ncdf4::ncatt_get(nc, "latitude", "units")[["value"]] not equal to "degrees_north".
# 1/1 mismatches
# x[1]: "degree_north"
# y[1]: "degrees_north"
# * ncdf4::ncatt_get(nc, "longitude", "units")[["value"]] not equal to "degrees_east".
# 1/1 mismatches
# x[1]: "degree_east"
# y[1]: "degrees_east"
# 2018-07-26 10:40:07 SEVERE [check_met_input.R#62: PEcAn.logger::logger.severe] : 
#    At least one error was found. 
# Error in PEcAn.logger::logger.severe("At least one error was found.") : 
#   At least one error was found.
```

I like this output (1) the context of the error is clear ("Dimensions are correct"), and (2) all the errors show up at once, with informative error messages.